src/anomalib/utils/metrics/anomaly_score_threshold.py

```diff
@@ -60,3 +60,51 @@ def compute(self) -> Tensor:
         else:
             self.value = thresholds[torch.argmax(f1_score)]
         return self.value
+
+
+class MeanVarianceMode(str, Enum):
+    """Supported Mean Variance Combination Type"""
+
+    ADD = "add"
+    MINUS = "minus"
+
+
+class AnomalyScoreMeanVarThreshold(AnomalyScoreThreshold):
+    """Anomaly Score Threshold.
+
+    This class computes/stores the threshold that determines the anomalous label
+    given anomaly scores. The computation is based on the mean and variance.
+
+    """
+
+    STD_DEV_RATIO = 0.1
+    MEANVAR_MODE = MeanVarianceMode.ADD
+
+    def __init__(self, default_value: float = 0.5, **kwargs) -> None:
+        self.meanvar_mode = self.MEANVAR_MODE
+        self.std_dev_ratio = self.STD_DEV_RATIO
+        super().__init__(default_value=default_value, **kwargs)
+
+    def compute(self) -> Tensor:
+        """Compute the threshold.
+
+        Compute the threshold based on the mean and variance of the predictions
+
+        Returns:
+            Value of threshold.
+        """
+
+        if not any(1 in batch for batch in self.target):
+            warnings.warn(
+                "The validation set does not contain any anomalous images. As a result, the adaptive threshold will "
+                "take the value of the highest anomaly score observed in the normal validation images, which may lead "
+                "to poor predictions. For a more reliable adaptive threshold computation, please add some anomalous "
+                "images to the validation set."
+            )
+        mean_pred = torch.mean(torch.cat(self.preds, dim=0))
+        std_pred = torch.std(torch.cat(self.preds, dim=0))
+        if self.meanvar_mode == MeanVarianceMode.MINUS:
+            self.value = mean_pred - self.std_dev_ratio * std_pred
```

Comment:
As you mentioned in your description this does not make sense in the anomaly context. Maybe we shouldn't support `minus` operation.

Reply:
We may add it as a placeholder. If in the future, if we use negative outputs of the model, then minus mode can be used?